src/UniTask/Assets/Plugins/UniTask/Runtime/External/DOTween/DOTweenAsyncExtensions.cs

```diff
@@ -250,30 +251,7 @@ public static IUniTaskSource Create(Tween tween, TweenCancelBehaviour cancelBeha
                                 source.tween.Complete(true);
                                 break;
                             case TweenCancelBehaviour.CancelAwait:
-                                // restore to original callback
-                                switch (callbackType)
-                                {
-                                    case CallbackType.Kill:
-                                        tween.onKill = source.originalCompleteAction;
-                                        break;
-                                    case CallbackType.Complete:
-                                        tween.onComplete = source.originalCompleteAction;
-                                        break;
-                                    case CallbackType.Pause:
-                                        tween.onPause = source.originalCompleteAction;
-                                        break;
-                                    case CallbackType.Play:
-                                        tween.onPlay = source.originalCompleteAction;
-                                        break;
-                                    case CallbackType.Rewind:
-                                        tween.onRewind = source.originalCompleteAction;
-                                        break;
-                                    case CallbackType.StepComplete:
-                                        tween.onStepComplete = source.originalCompleteAction;
-                                        break;
-                                    default:
-                                        break;
-                                }
+                                source.RestoreOriginalCallback();
```

Comment:
>  // restore to original callback

I made this a method  (because duplicated). There should be no change in behavior.